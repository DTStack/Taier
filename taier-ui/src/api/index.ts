import http from './http';
import req from './request';
import { TASK_TYPE_ENUM } from '@/constant';

export default {
    // 获取节点下拉
    getNodeAddressSelect(params?: any) {
        return http.post(req.GET_NODEADDRESS_SELECT, params);
    },
    // 获取类型数据源
    getTypeOriginData(params: any) {
        return http.post(req.GET_TYPE_ORIGIN_DATA, params);
    },
    listTablesBySchema(params: any) {
        return http.post(req.LIST_TABLE_BY_SCHEMA, params);
    },
    // 获取kafka topic预览数据
    getDataPreview(params: any) {
        return http.post(req.GET_DATA_PREVIEW, params);
    },
    pollPreview(params: any) {
        return http.post(req.POLL_PREVIEW, params);
    },
    // 添加或更新任务
    saveTask(params: any) {
        return http.post(req.SAVE_TASK, params);
    },
    // 获取Topic
    getTopicType(params: any) {
        return http.post(req.GET_TOPIC_TYPE, params);
    },
    getStreamTableColumn(params: { schema: string; sourceId: number; tableName: string; flinkVersion: string }) {
        return http.post(req.GET_STREAM_TABLECOLUMN, params);
    },
    // 获取源表中的时区列表
    getTimeZoneList(params?: any) {
        return http.post(req.GET_TIMEZONE_LIST, params);
    },
    // 转换向导到脚本模式
    convertToScriptMode(params: any) {
        return http.post(req.CONVERT_TO_SCRIPT_MODE, params);
    },
    checkSyntax(params: any) {
        return http.post(req.GRAMMAR_CHECK, params);
    },
    sqlFormat(params: any) {
        return http.post(req.SQL_FORMAT, params);
    },
    getTaskList(params: any) {
        return http.post(req.GET_TASK_LIST, params);
    },
    getStatusCount(params: any) {
        return http.post(req.GET_STATUS_COUNT, params);
    },
    startTask(params: any) {
        return http.post(req.START_TASK, params);
    },
    startCollectionTask(params: any) {
        return http.post(req.START_COLLECTION_TASK, params);
    },
    getTaskManagerLog(params: any) {
        return http.post(req.GET_TASK_MANAGER_LOG, params);
    },
    getJobManagerLog(params: any) {
        return http.post(req.GET_JOB_MANAGER_LOG, params);
    },
    listTaskManager(params: any) {
        return http.post(req.LIST_TASK_MANAGER, params);
    },
    getTaskLogs(params: any) {
        return http.post(req.GET_TASK_LOGS, params);
    },
    // failover 日志
    getFailoverLogsByTaskId(params: any) {
        return http.post(req.GET_TASK_FAILOVER_LOG, params);
    },
    getHistoryLog(params: any) {
        return http.post(req.GET_HISTORY_LOG, params);
    },
    isOpenCdb(params: { dataInfoId: number }) {
        return http.post(req.IS_OPEN_CDB, params);
    },
    getPDBList(params: { dataInfoId: number; searchKey?: string }) {
        return http.post(req.GET_PDB_LIST, params);
    },
    // 数据开发 - 获取启停策略列表
    getAllStrategy() {
        return http.post(req.GET_ALL_STRATEGY);
    },
    getTopicPartitionNum(params: any) {
        return http.post(req.GET_TOPIC_PARTITION_NUM, params);
    },
    getSchemaTableColumn(params: any) {
        return http.post(req.GET_SCHEMA_TABLE_COLUMN, params);
    },
    getSlotList(params: any) {
        return http.post(req.GET_SLOT_LIST, params);
    },
    getBinlogListBySource(params: any) {
        return http.post(req.GET_BINLOG_LIST_BY_SOURCE, params);
    },
    // 获取指标
    getTaskMetrics(params: { taskId: number; timespan: string; end: number; chartNames: string[] }) {
        return http.post(req.GET_TASK_METRICS, params);
    },
    getMetricValues(params: { taskId: number }) {
        // 获取所有指标
        return http.get(req.GET_METRIC_VALUES, params);
    },
    checkSourceStatus(params: { taskId: number }) {
        // 获取任务的异常数据源
        return http.post(req.CHECK_SOURCE_STATUS, params);
    },
    queryTaskMetrics(params: { taskId: number; chartName: string; timespan: string; end: number }) {
        // 查询指标数据
        return http.post(req.QUERY_TASK_METRICES, params);
    },
    getListHistory(params: any) {
        return http.post(req.GET_LIST_HISTORY, params);
    },
    listCheckPoint(params: any) {
        return http.post(req.LIST_CHECK_POINT, params);
    },
    stopTask(params: any) {
        return http.post(req.STOP_TASK, params);
    },
    getTaskJson(params: any) {
        return http.post(req.GET_TASK_JSON, params);
    },
    getTaskSqlText(params: any) {
        return http.post(req.GET_TASK_SQL_TEXT, params);
    },
    addTenant(params: any) {
        return http.post(req.ADD_TENANT, params);
    },
    switchTenant(params: { tenantId: number }) {
        return http.post(req.SWITCH_TENANT, params);
    },
    login(params: { username: string; password: string }) {
        return http.postAsFormData(req.LOGIN, params);
    },
    addCluster(params: { clusterName: string }) {
        return http.post(req.ADD_CLUSTER, params); // 新增集群
    },
    getClusterInfo(params: { clusterId: number | string }) {
        return http.get(req.GET_CLUSTER_INFO, params);
    },
    uploadResource(params: { fileName: any; componentType: any }) {
        return http.postAsFormData(req.UPLOAD_RESOURCE, params);
    },
    deleteComponent(params: { componentId: number }) {
        return http.post(req.DELETE_COMPONENT, params); // 删除组件
    },
    deleteCluster(params: { clusterId: number }) {
        return http.post(req.DELETE_CLUSTER, params);
    },
    testConnect(params: { clusterId: number; componentType: number; versionName: string }) {
        return http.post(req.TEST_CONNECT, params);
    },
    testConnects<T>(params: { clusterId: number }) {
        return http.post<T>(req.TEST_CONNECTS, params);
    },
    closeKerberos(params: { componentId: number }) {
        return http.post(req.CLOSE_KERBEROS, params);
    },
    getVersionData(params?: any) {
        return http.get(req.GET_VERSION, params);
    },
    saveComponent(params: any) {
        return http.postAsFormData(req.SAVE_COMPONENT, params);
    },
    parseKerberos(params: any) {
        return http.postAsFormData(req.PARSE_KERBEROS, params);
    },
    getClusterList(params: { currentPage: number; pageSize: number }) {
        return http.post(req.GET_CLUSTER_LIST, params);
    },
    getTenantList(params?: any) {
        return http.get(req.GET_TENANT_LIST, params);
    },
    getMetaComponent(params?: any) {
        return http.get(req.GET_META_COMPONENT, params);
    },
    // 获取存储组件列表
    getComponentStore(params: any) {
        return http.post(req.GET_COMPONENTSTORE, params);
    },
    // 上传kerberos文件
    uploadKerberos(params: any) {
        return http.postAsFormData(req.UPLOAD_KERBEROS, params);
    },
    // 更新krb5.conf文件
    updateKrb5Conf(params: { krb5Content: string }) {
        return http.post(req.UPDATE_KRB5CONF, params);
    },
    // 概览-获取集群
    getClusterDetail(params: any) {
        return http.post(req.GET_CLUSTER_DETAIL, params);
    },
    // 明细-杀死选中或者杀死全部任务
    killTasks(params: { jobResource: string; nodeAddress: string; stage: number; jobIdList?: any[] }) {
        return http.post(req.KILL_TASKS, params);
    },
    killAllTask(params: { jobResource: string; nodeAddress: string }) {
        return http.post(req.KILL_ALL_TASK, params);
    },
    stickJob(params: { jobId: string; jobResource: string }) {
        return http.post(req.JOB_STICK, params);
    },
    // 查看明细 和搜索条件
    getViewDetail(params: {
        stage: number;
        jobResource: string;
        nodeAddress?: string;
        currentPage: number;
        pageSize: number;
    }) {
        return http.post(req.GET_VIEW_DETAIL, params);
    },
    getClusterResources(params: any) {
        return http.post(req.GET_CLUSTER_RESOURCES, params);
    },
    getLoadTemplate<T>(params: any) {
        return http.post<T>(req.GET_LOADTEMPLATE, params);
    },
    getAllCluster(params?: any) {
        return http.get(req.GET_ALL_CLUSTER, params);
    },
    getEnginesByCluster(params?: any) {
        return http.get(req.GET_ENGINES_BY_CLUSTER, params);
    },
    searchTenant(params: any) {
        return http.post(req.SEARCH_TENANT, params);
    },
    bindTenant(params: any) {
        return http.post(req.BIND_TENANT, params);
    },
    switchQueue(params: any) {
        return http.post(req.SWITCH_QUEUE, params);
    },
    refreshQueue(params: { clusterName: string }) {
        return http.post(req.REFRESH_QUEUE, params);
    },
    getRetainDBList(params?: any) {
        return http.post(req.GET_RETAINDB_LIST, params);
    },
    convertToHiveColumns(params: any) {
        return http.post(req.CONVERT_TO_HIVE_COLUMNS, params);
    },
    convertDataSyncToScriptMode(params: any) {
        return http.post(req.CONVERT_SYNC_T0_SCRIPT_MODE, params);
    },
    getOfflineTaskByID<T = any>(params: any) {
        return http.post<T>(req.GET_TASK, params);
    },
    getOfflineSubTaskById<T = any>(params: any) {
        return http.post<T>(req.GET_SUB_TASK, params);
    },
    getCustomParams(params?: any) {
        return http.post(req.GET_CUSTOM_TASK_PARAMS, params);
    },
    getSyncTemplate(params: any) {
        return http.post(req.GET_SYNC_SCRIPT_TEMPLATE, params);
    },
    publishOfflineTask(params: any) {
        return http.post(req.PUBLISH_TASK, params);
    },
    getTaskTypes(params?: any) {
        return http.post(req.GET_TASK_TYPES, params);
    },
    getTableInfoByDataSource(params: any) {
        return http.post(req.GET_TABLE_INFO_BY_DATASOURCE, params);
    },
    execSQLImmediately<T>(params: any) {
        // 立即执行SQL
        return http.post<T>(req.EXEC_SQL_IMMEDIATELY, params);
    },
    stopSQLImmediately(params: any) {
        // 停止执行数据同步
        return http.post(req.STOP_SQL_IMMEDIATELY, params);
    },
    execDataSyncImmediately<T>(params: any) {
        // 立即执行数据同步
        return http.post<T>(req.EXEC_DATA_SYNC_IMMEDIATELY, params);
    },
    stopDataSyncImmediately(params: any) {
        // 停止执行SQL
        return http.post(req.STOP_DATA_SYNC_IMMEDIATELY, params);
    },
    getIncrementColumns(params: any, config?: any) {
        // 获取增量字段
        return http.post(req.GET_INCREMENT_COLUMNS, params, config);
    },
    getHivePartitions(params: any) {
        // 获取Hive分区
        return http.post(req.CHECK_HIVE_PARTITIONS, params);
    },
    selectRunLog<T>(params: any) {
        // 非数据同步接口获取日志
        return http.post<T>(req.SELECT_SQL_LOG, params);
    },
    /**
     * - 查询数据同步任务，SQL 执行结果
     * - 需要补充增量同步
     * @param {Object} params 请求参数
     * @param {Number} taskType 任务类型
     */
    selectExecResultData(params: any, taskType: any) {
        //
        const url =
            taskType && taskType === TASK_TYPE_ENUM.SYNC ? req.SELECT_DATA_SYNC_RESULT : req.SELECT_SQL_RESULT_DATA;
        return http.post(url, params);
    },
    selectExecResultDataSync(params: any) {
        // 数据同步接口获取结果表
        return http.post(req.SELECT_DATA_SYNC_RESULT, params);
    },
    selectStatus(params: any) {
        // 非数据同步接口轮训状态
        return http.post(req.SELECT_SQL_STATUS, params);
    },
    forzenTask(params: any) {
        return http.post(req.FROZEN_TASK, params);
    },
    getOfflineCatalogue(params: any) {
        return http.post(req.GET_OFFLINE_CATALOGUE, params);
    },
    addOfflineCatalogue(params: any) {
        return http.post(req.ADD_OFFLINE_CATALOGUE, params);
    },
    editOfflineCatalogue(params: any) {
        return http.post(req.EDIT_OFFLINE_CATALOGUE, params);
    },
    addOfflineResource(params: any) {
        return http.postAsFormData(req.ADD_OFFLINE_RESOURCE, params);
    },
    replaceOfflineResource(params: any) {
        return http.postAsFormData(req.REPLACE_OFFLINE_RESOURCE, params);
    },
    addOfflineTask(params: any) {
        return http.post(req.ADD_OFFLINE_TASK, params);
    },
    saveOfflineJobData(params: any) {
        return http.post(req.SAVE_OFFLINE_JOBDATA, params);
    },
    editTask(params: any) {
        return http.post(req.EDIT_TASK, params);
    },
    addOfflineFunction(params: any) {
        return http.post(req.ADD_OFFLINE_FUNCTION, params);
    },
    delOfflineTask(params: any) {
        return http.post(req.DEL_OFFLINE_TASK, params);
    },
    delOfflineFolder(params: any) {
        return http.post(req.DEL_OFFLINE_FOLDER, params);
    },
    delOfflineRes(params: any) {
        return http.post(req.DEL_OFFLINE_RES, params);
    },
    delOfflineFn(params: any) {
        return http.post(req.DEL_OFFLINE_FN, params);
    },
    getOfflineFn(params: any) {
        return http.post(req.GET_FN_DETAIL, params);
    },
    getOfflineRes(params: any) {
        return http.post(req.GET_RES_DETAIL, params);
    },
    getHBaseColumnFamily(params: any) {
        return http.post(req.GET_HBASE_COLUMN_FAMILY, params);
    },
    queryOfflineTasks(params: any) {
        return http.post(req.QUERY_TASKS, params);
    },
    getOfflineTaskLog(params: any) {
        // 获取离线任务日志
        return http.post(req.GET_TASK_LOG, params);
    },
    getOfflineTaskPeriods(params: any) {
        // 转到前后周期
        return http.post(req.GET_TASK_PERIODS, params);
    },
    queryJobs<T = any>(params: any) {
        return http.post<T>(req.QUERY_JOBS, params);
    },
    getSubJobs<T = any>(params: any) {
        return http.post<T>(req.GET_SUB_JOBS, params);
    },
    patchTaskData(params: any) {
        // 补数据
        return http.post(req.PATCH_TASK_DATA, params);
    },
    getTaskChildren<T = any>(params: any) {
        // 获取任务子节点
        return http.post<T>(req.GET_TASK_CHILDREN, params);
    },
    getRootWorkflowTask<T = any>(params: any) {
        return http.post<T>(req.GET_ROOT_WORKFLOW_TASK, params);
    },
    getFillData(params: any) {
        // 补数据搜索
        return http.post(req.GET_FILL_DATA, params);
    },
    getFillDataDetail<T = any>(params: any) {
        // 补数据详情
        return http.post<T>(req.GET_FILL_DATA_DETAIL, params);
    },

    batchStopJob(params: any) {
        // 批量停止任务
        return http.post(req.BATCH_STOP_JOBS, params);
    },
    batchRestartAndResume(params: any) {
        // 重启并恢复任务
        return http.post(req.BATCH_RESTART_AND_RESUME_JOB, params);
    },
    getJobChildren<T = any>(params: any) {
        // 获取任务子Job
        return http.post<T>(req.GET_JOB_CHILDREN, params);
    },
    getRootWorkflowJob<T = any>(params: any) {
        return http.post<T>(req.GET_ROOT_WORKFLOW_JOB, params);
    },
    queryJobStatics(params: any) {
        return http.post(req.QUERY_JOB_STATISTICS, params);
    },

    stopFillDataJobs(params: any) {
        return http.post(req.STOP_FILL_DATA_JOBS, params);
    },

    getPersonInCharge() {
        return http.post(req.USER_QUERYUSER);
    },
    /**
     * 获取工作流任务节点实例的子节点
     */
    getTaskJobWorkflowNodes(params: any) {
        return http.post(req.GET_TASK_JOB_WORKFLOW_NODES, params);
    },
    getOfflineTableList(params: any, config?: any) {
        return http.post(req.TABLE_LIST, params, config);
    },
    getOfflineTableColumn(params: any, config?: any) {
        return http.post(req.GET_TABLE_COLUMN, params, config);
    },
    getOfflineColumnForSyncopate(params: any, config?: any) {
        return http.post(req.GET_COLUMN_FOR_SYNCOPATE, params, config);
    },
    getHivePartitionsForDataSource(params: any, config?: any) {
        return http.post(req.GET_HIVE_PARTITIONS, params, config);
    },
    getDataSourcePreview(params: any) {
        return http.post(req.GET_DATA_SOURCE_PREVIEW, params);
    },
    getAllSchemas(params: any, config?: any) {
        return http.post(req.GET_ALL_SCHEMAS, params, config);
    },
    dataSourcepage(params: any) {
        return http.post(req.GET_DATA_SOURCE_PAGE, params);
    },
    typeList(params: any) {
        return http.post(req.GET_TYPE_LIST, params);
    },
    dataSourceDelete(params: any) {
        return http.post(req.DELETE_SOURCE, params);
    },
    queryDsClassifyList(params: any) {
        return http.post(req.QUERY_DATA_SOURCE_CLASSIFY_LIST, params);
    },
    queryDsTypeByClassify(params: any) {
        return http.post(req.QUERY_LIST_BY_CLASSIFY, params);
    },
    queryDsVersionByType(params: any) {
        return http.post(req.QUERY_VERSION_BY_CLASSIFY, params);
    },
    findTemplateByTypeVersion(params: any) {
        return http.post(req.QUERY_TEMPLATE_BY_VERSION, params);
    },
    addDatasource(params: any) {
        return http.post(req.ADD_DATA_SOURCE, params);
    },
    addOrUpdateSourceWithKerberos(params: any) {
        return http.postForm(req.UPLOAD_DATA_SOURCE_WITH_KERBEROS, params);
    },
    testCon(params: any) {
        return http.post(req.TEST_CONNECT_IN_DATA_SOURCE, params);
    },
    testConWithKerberos(params: any) {
        return http.postForm(req.TEST_KERBEROS_IN_DATA_SOURCE, params);
    },
    detail(params: any) {
        return http.post(req.GET_DATA_SOURCE_DETAIL, params);
    },
    uploadCode(params: any) {
        return http.postForm(req.UPLOAD_CODE, params);
    },
    getCreateTargetTable(params: any) {
        return http.post(req.GET_CREATE_TARGET_TABLE, params);
    },
    createDdlTable(params: any) {
        return http.post(req.CREATE_DDL_TABLE, params);
    },
    batchStopJobByDate(params: any) {
        // 按业务日期批量杀任务
        return http.post(req.BATCH_STOP_JOBS_BY_DATE, params);
    },
    allProductGlobalSearch(params: any) {
        return http.post(req.ALL_PRODUCT_GLOBAL_SEARCH, params);
    },
    getComponentModels() {
        return http.get(req.GET_COMPONENT_MODELS, {});
    },
    getComponentInfo(params: any) {
        return http.get(req.GET_COMPONENT_INFO, params);
    },
    getComponentSchemaConfig<T>(params: any) {
        return http.post<T>(req.GET_TENANT_COMPONENT_LIST, params);
    },
    saveComponentSchemaConfig(params: any) {
        return http.post(req.SAVE_TENANT_COMPONENT_INFO, params);
    },
    getSchemaListByComponent<T>(params: any) {
        return http.post<T>(req.GET_SCHEMA_LIST_BY_COMPONENT, params);
    },
    getResourceLocation<T = any>(params: any) {
        return http.post<T>(req.GET_RESOUCE_LOCATION, params);
    },
    validateRepeatTaskName<T = any>(params: any) {
        return http.post<T>(req.VALIDATE_REPEAT_TASK_NAME, params);
    },
    getSyncProperties<T = any>(params: any) {
        return http.post<T>(req.QUERY_SYNC_DATA_SOURCE, params);
    },
    getSupportSource<T = any>(params: any) {
        return http.post<T>(req.QUERY_SUPPORT_SOURCE, params);
    },
    getResourceByTenant(params: any) {
        return http.post(req.GET_RESOURCES_BY_TENANT, params);
    },
    getAllDataSource(params: any) {
        return http.post(req.GET_ALL_DATA_SOURCE, params);
    },
    getComponentVersionByTaskType<T = any>(params: any) {
        return http.post<T>(req.GET_COMPONENT_VERSION, params);
    },
    getFTPColumns<T = any>(params: any) {
        return http.post<T>(req.GET_FTP_COLUMNS, params);
    },
};
