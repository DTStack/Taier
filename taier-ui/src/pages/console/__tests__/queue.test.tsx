import api from '@/api';
import { $, $$, selectItem, toggleOpen, triggerOkOnConfirm } from '@/tests/utils';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import Queue from '../queue';
import '@testing-library/jest-dom';
import { history } from 'umi';
import { getAllCluster, getClusterDetail, getNodeAddressSelect } from './fixtures/mock';

jest.mock('@/api');
jest.useFakeTimers();
// use the original Modal
jest.mock('antd', () => {
    const original = jest.requireActual('antd');
    return {
        ...original,
    };
});
jest.mock('umi', () => ({
    history: {
        push: jest.fn(),
    },
}));

describe('Test Queue Page', () => {
    beforeEach(() => {
        (api.getAllCluster as jest.Mock).mockReset().mockResolvedValue(getAllCluster);
        (api.getNodeAddressSelect as jest.Mock).mockReset().mockResolvedValue(getNodeAddressSelect);
        (api.getClusterDetail as jest.Mock).mockReset().mockResolvedValue(getClusterDetail);
    });

    it('Should match snapshot', async () => {
        const { asFragment } = render(<Queue />);

        await waitFor(() => {
            expect($$('.ant-table-row').length).toBe(2);
            expect(asFragment()).toMatchSnapshot();
        });
    });

    it('Should re-request', async () => {
        render(<Queue />);

        // Get data at first time
        await waitFor(() => {
            expect(api.getClusterDetail).toBeCalledTimes(1);
        });

        // Get data since the refresh button clicked
        fireEvent.click($('button.dt-refresh')!.firstChild!);
        await waitFor(() => {
            expect(api.getClusterDetail).toBeCalledTimes(2);
        });

        // Get data since the cluster changed
        toggleOpen($$<HTMLDivElement>('.ant-form-item')[0]!);
        await act(async () => {
            selectItem(1);
        });
        await waitFor(() => {
            expect(api.getClusterDetail).toBeCalledTimes(3);
        });

        // Get data since the node changed
        toggleOpen($$<HTMLDivElement>('.ant-form-item')[1]!);
        await act(async () => {
            selectItem(0, 1);
        });
        await waitFor(() => {
            expect(api.getClusterDetail).toBeCalledTimes(4);
        });
    });

    it('Should support kill jobs', async () => {
        (api.killAllTask as jest.Mock).mockReset().mockResolvedValue({
            code: 1,
        });
        const { getAllByText, findByText } = render(<Queue />);
        await waitFor(() => {
            expect($$('.ant-table-row').length).toBe(2);
        });

        fireEvent.click(getAllByText('杀死全部')[0]);

        expect(
            await findByText('杀死全部', {
                selector: 'span.ant-modal-confirm-title',
            })
        ).toBeInTheDocument();

        await triggerOkOnConfirm();

        await waitFor(() => {
            expect(api.killAllTask).toBeCalled();
        });
    });

    it('History should push', async () => {
        const { findByText } = render(<Queue />);

        expect(await findByText('spark_sql_default_default_batch_Yarn', { selector: 'a' })).toBeInTheDocument();

        fireEvent.click(await findByText('spark_sql_default_default_batch_Yarn', { selector: 'a' }));

        expect(history.push).toBeCalledWith({
            query: {
                clusterName: 'default',
                drawer: 'queue-detail',
                jobResource: 'spark_sql_default_default_batch_Yarn',
                jobStage: '2',
                node: undefined,
            },
        });
    });
});
