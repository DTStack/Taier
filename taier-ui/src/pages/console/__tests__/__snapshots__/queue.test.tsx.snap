// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Queue Page Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="dt-sketch dt-queue"
  >
    <div
      class="dt-sketch-header"
    >
      <form
        autocomplete="off"
        class="ant-form ant-form-inline"
      >
        <div
          class="ant-form-item ant-form-item-has-success"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-label"
            >
              <label
                class=""
                for="clusterId"
                title="集群"
              >
                集群
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-select ant-select-in-form-item ant-select-status-success ant-select-single ant-select-show-arrow"
                    style="width: 150px;"
                  >
                    <div
                      class="ant-select-selector"
                    >
                      <span
                        class="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="clusterId_list_0"
                          aria-autocomplete="list"
                          aria-controls="clusterId_list"
                          aria-haspopup="listbox"
                          aria-owns="clusterId_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="clusterId"
                          readonly=""
                          role="combobox"
                          style="opacity: 0;"
                          type="search"
                          unselectable="on"
                          value=""
                        />
                      </span>
                      <span
                        class="ant-select-selection-item"
                        title="default"
                      >
                        default
                      </span>
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-arrow"
                      style="user-select: none;"
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        class="anticon anticon-down ant-select-suffix"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-label"
            >
              <label
                class=""
                for="clusterNode"
                title="节点"
              >
                节点
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-select ant-select-in-form-item ant-select-single ant-select-allow-clear ant-select-show-arrow"
                    style="width: 150px;"
                  >
                    <div
                      class="ant-select-selector"
                    >
                      <span
                        class="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="clusterNode_list_0"
                          aria-autocomplete="list"
                          aria-controls="clusterNode_list"
                          aria-haspopup="listbox"
                          aria-owns="clusterNode_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="clusterNode"
                          readonly=""
                          role="combobox"
                          style="opacity: 0;"
                          type="search"
                          unselectable="on"
                          value=""
                        />
                      </span>
                      <span
                        class="ant-select-selection-placeholder"
                      >
                        选择节点
                      </span>
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-arrow"
                      style="user-select: none;"
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        class="anticon anticon-down ant-select-suffix"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
      <div
        class="dt-sketch-extra"
      >
        <button
          class="ant-btn ant-btn-default dt-refresh"
          type="button"
        >
          <span
            aria-label="sync"
            class="anticon anticon-sync"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="sync"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="dt-sketch-header-bar"
    />
    <div
      class="ant-table-wrapper dt-sketch-table"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
              >
                <table
                  style="table-layout: fixed;"
                >
                  <colgroup>
                    <col
                      style="width: 180px;"
                    />
                  </colgroup>
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell ant-table-cell-ellipsis"
                        title="计算类型"
                      >
                        计算类型
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        队列中(等待时长)
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        已存储
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        等待重试
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        等待资源
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        运行中
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        操作
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="spark_sql_default_default_batch_Yarn"
                    >
                      <td
                        class="ant-table-cell ant-table-cell-ellipsis"
                      >
                        <a>
                          spark_sql_default_default_batch_Yarn
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          0
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          0
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          0
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          25
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          0
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          杀死全部
                        </a>
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="shell_default_default_batch_"
                    >
                      <td
                        class="ant-table-cell ant-table-cell-ellipsis"
                      >
                        <a>
                          shell_default_default_batch_
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          0
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          0
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          0
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          7
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          0
                        </a>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a>
                          杀死全部
                        </a>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
